crux-core/src/crux/api/v2/DeleteOperation.java

```diff
@@ -0,0 +1,32 @@
+package crux.api.v2;
+
+import clojure.lang.Keyword;
+import clojure.lang.PersistentVector;
+
+import java.util.Date;
+
+public class DeleteOperation extends Operation {
+    private final Date validTime;
+    private final CruxId deleteId;
+
+    private DeleteOperation(CruxId deleteId, Date validTime) {
+        this.deleteId = deleteId;
+        this.validTime = validTime;
+    }
+
+    public DeleteOperation withValidTime(Date validTime) {
+        return new DeleteOperation(deleteId, validTime);
+    }
+
+    public static DeleteOperation deleteOp(CruxId deleteId) {
```

Comment:
A general comment here - in Java it's not common to see many abbreviations like `op` or `attr` - autocomplete eliminates the need to type or remember such names. The call will look like 
`DeleteOperation.deleteOp( id )`, which is a bit verbose (even by Java standards). It's rare to use static imports for such methods, so some people might prefer `DeleteOperation.create(idToDelete)`

Reply:
yep, agreed with the abbrev's ;) 

re the static imports - I wasn't aware that this was that rare? intention behind it was to make the name of the method self-contained, although this may be a result of us coming from the Clojure world where such things are common :)

in the meantime, I've moved these under `TransactionOperation`, so they can be used either fully qualified or by a static import